File: lcr/148/main.py
from typing import List


class Solution:
    def validateBookSequences(self, putIn: List[int], takeOut: List[int]) -> bool:
        i = 0
        stack = []

        for num in putIn:
            stack.append(num)

            while stack and stack[-1] == takeOut[i]:
                stack.pop()
                i += 1

        return not stack

File: lcr/148/main.go
package main

func validateBookSequences(putIn []int, takeOut []int) bool {
	i := 0
	stack := []int{}

	for _, num := range putIn {
		stack = append(stack, num)

		for len(stack) > 0 && stack[len(stack)-1] == takeOut[i] {
			stack = stack[:len(stack)-1]
			i += 1
		}
	}
	return len(stack) == 0
}
